metrics/nested_controlflows.py

```diff
@@ -8,10 +10,10 @@ def __init__(self):
         self._source_repository: SourceRepository = None
 
     def run(self):
-        data = []
+        result = Result("nested controlflow", [])
         for file in self._source_repository.src_paths:
-            self._parse(self._source_repository.getAst(file), file, data)
-        return data
+            self._parse(self._source_repository.getAst(file), file, result)
```

Comment:
I would return the `Violation` from `_parse` and then append to result in the for loop, instead of an implicit return through the `result` argument.